packages/node/src/login/oidc/oidcHandlers/AuthorizationCodeWithPkceOidcHandler.spec.ts

```diff
@@ -147,5 +149,65 @@ describe("AuthorizationCodeWithPkceOidcHandler", () => {
         mockedStorage.getForUser(oidcOptions.sessionId, "dpop"),
       ).resolves.toBe("false");
     });
+
+    it("emits AUTHORIZATION_REQUEST_STATE event with codeVerifier and state when eventEmitter is provided", async () => {
+      const mockedStorage = mockStorageUtility({});
+      const eventEmitter = new EventEmitter();
+      const authStateSpy = jest.fn();
+      eventEmitter.on(EVENTS.AUTHORIZATION_REQUEST_STATE, authStateSpy);
+
+      const authorizationCodeWithPkceOidcHandler =
+        getAuthorizationCodeWithPkceOidcHandler({
+          storageUtility: mockedStorage,
+        });
+      const oidcOptions = mockDefaultOidcOptions();
+      oidcOptions.eventEmitter = eventEmitter;
+
+      await authorizationCodeWithPkceOidcHandler.handle(oidcOptions);
+
+      expect(authStateSpy).toHaveBeenCalledTimes(1);
+      expect(authStateSpy).toHaveBeenCalledWith(
+        expect.objectContaining({
+          codeVerifier: expect.any(String),
+          state: expect.any(String),
+        }),
+      );
+
+      const authStateArg = authStateSpy.mock.calls[0][0] as {
+        state: string;
+        codeVerifier: string;
+      };
```

Comment:
```suggestion
      const authStateArg = authStateSpy.mock.calls[0][0];
```
The type of `authStateArg` is inferred from the mock type.